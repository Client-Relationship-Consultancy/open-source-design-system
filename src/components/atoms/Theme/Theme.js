import React from "react"
import PropTypes from "prop-types"
import { ThemeProvider } from "styled-components"
import { colourPalette } from "../../../brandColours.ts"

const Theme = props => (
  <ThemeProvider theme={props.theme || colourPalette.examplePalette}>
    <>{props.children}</>
  </ThemeProvider>
)

Theme.defaultProps = {
  theme: colourPalette.examplePalette,
}

Theme.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node,
}

export default Theme
