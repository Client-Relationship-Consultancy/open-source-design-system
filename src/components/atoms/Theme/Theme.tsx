import React from "react"
import { ThemeProvider } from "styled-components"
import { colourPalette, IColourPalette } from "../../../brandColours"

interface IProps{
  theme: IColourPalette;
}


const Theme: React.FunctionComponent<IProps> = props => (
  <ThemeProvider theme={props.theme || colourPalette.examplePalette}>
    <>{props.children}</>
  </ThemeProvider>
)


export default Theme
