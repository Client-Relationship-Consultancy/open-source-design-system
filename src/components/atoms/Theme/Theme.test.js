import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import Theme from "./Theme"
import "jest-styled-components"

describe("List Items Testing", () => {
  const component = shallow(<Theme />)
  it("Snapshot Testing", () => {
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
