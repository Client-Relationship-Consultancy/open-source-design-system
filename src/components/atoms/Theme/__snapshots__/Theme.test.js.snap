// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Items Testing Snapshot Testing 1`] = `
<ThemeProvider
  theme="examplePalette"
/>
`;
